src/main/java/com/researchspace/dcd/client/DigitalCommonsDataClient.java

```diff
@@ -0,0 +1,43 @@
+package com.researchspace.dcd.client;
+
+import com.researchspace.dcd.model.DigitalCommonDataSubmission;
+import com.researchspace.dcd.model.DigitalCommonsDataDataset;
+import com.researchspace.dcd.model.DigitalCommonsDataFile;
+import java.io.File;
+import java.io.IOException;
+
+/**
+ * API Client wrapper for making calls to DMP API.
+ */
+public interface DigitalCommonsDataClient {
+
+	/**
+	 * Create a DigitalCommonsData Deposition.
+	 *
+	 * A DigitalCommonsData Dataset is used for editing and uploading records to DigitalCommonsData
+	 * and acts as a conceptual grouping of files. Here, we provide two methods
+	 * for creating a new Deposition; one with predefined metadata and one
+	 * without. In either case, a new Dataset is created on the DigitalCommonsData
+	 * server and its details are returned.
+	 *
+	 * @param submission The metadata to associate with the Deposition.
+	 * @return The created DigitalCommonsData deposition.
+	 */
+	DigitalCommonsDataDataset createDataset(DigitalCommonDataSubmission submission) throws IOException;
+
+	Boolean deleteDataset(String datasetId);
+
+	Boolean isConnectionOpen();
```

Comment:
Wouldn't better name for the method be something like `testConnection()`? That'd match usage in adapter, and also when using DCD API there is just a token that is valid or not, while the method name here suggest there is some concept of opened/closed connection (which we sometimes have, e.g. in external filestore session).

Reply:
yes I think the name you proposed is actually more appropriate, changed now